src/rendering/renderers/gpu/shader/GpuShaderSystem.ts

```diff
@@ -24,13 +25,17 @@ export class GpuShaderSystem implements System
         name: 'shader',
     } as const;
 
+    public maxTextures: number;
+
     private _gpu: GPU;
 
     private readonly _gpuProgramData: Record<number, GPUProgramData> = Object.create(null);
 
     protected contextChange(gpu: GPU): void
     {
         this._gpu = gpu;
+
+        this.maxTextures = getMaxTexturesPerBatch();
```

Comment:
I suppose we could use webgpu here to determine the maximum textures given the webgpu context. But I don't know how that would work.

Reply:
a good point, this turned out to be a pain as often max textures is required before a renderer is created. Figured WebGL would be a safest to just use in all cases!

Reply:
although here, we could do:
```gpu.device.limits.maxSampledTexturesPerShaderStage```

Reply:
Made the change to use `gpu.device.limits.maxSampledTexturesPerShaderStage`.